aoc/src/main/scala/aoc2015/Day06.scala

```diff
@@ -23,39 +23,27 @@ object Day06 extends App:
       .map(Instructions.fromString)
       .toList
 
-  val answer1: Int =
-    val grid: List[List[Boolean]] =
-      List.tabulate(1000,1000)((x,y) =>
+  def firstInstructions(light: Boolean, instructions: Instructions): Boolean =
+    instructions.cmd match
+      case "toggle" => !light
+      case "on" => true
+      case "off" => false
+
+  def secondInstructions(light: Int, instructions: Instructions): Int =
+    instructions.cmd match
+      case "toggle" => light + 2
+      case "on" => light + 1
+      case "off" => if light == 0 then light else light - 1
+
+  def execute[A](startVal: A, typeInstr: (A, Instructions) => A): List[List[A]] =
```

Comment:
wow - very, _very_ nice solution - actually much clearer than the one ive shown you

well done